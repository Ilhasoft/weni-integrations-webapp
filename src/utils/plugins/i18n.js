import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
