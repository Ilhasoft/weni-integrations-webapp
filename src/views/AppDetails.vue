<template>
  <div>
    <navigator class="navigator" :routes="navigatorHistory" />

    <div class="app-details" v-if="!loading">
      <app-images-banner class="app-details__section" :images="app.banners" />
      <app-details-header
        class="app-details__section"
        :title="app.name"
        :description="app.summary"
        :icon="app.icon"
        :iconbgColor="app.bg_color"
        :appCode="app.code"
      />
      <unnnic-banner
        class="app-details__banner"
        :key="bannerKey"
        type="info"
        :firstTitle="$t('apps.details.info')"
        :firstDescription="appMetrics"
        :secondTitle="$t('apps.details.integrated_into')"
        :secondDescription="appIntegrationsCount"
        :subtitle="$t('apps.details.organizations')"
        :thirdTitle="$t('apps.details.rating')"
        :thirdDescription="appRatingString"
        :rating="appRatingAverage"
        @ratingAction="handleRating"
      />
      <div class="app-details__section app-details__section__columns">
        <app-details-about :description="app.description" :links="appLinks" />
        <!-- <app-details-recommended class="app-details__section__columns__recommended" /> -->
      </div>
      <app-details-comments :appCode="app.code" />
    </div>
    <skeleton-loading v-else />
  </div>
</template>

<script>
  import Navigator from '../components/Navigator.vue';
  import AppImagesBanner from '../components/app/AppImagesBanner.vue';
  import AppDetailsHeader from '../components/app/AppDetailsHeader.vue';
  import AppDetailsAbout from '../components/app/AppDetailsAbout.vue';
  // import AppDetailsRecommended from '../components/app/AppDetailsRecommended.vue';
  import AppDetailsComments from '../components/app/AppDetailsComments.vue';
  import skeletonLoading from './loadings/AppDetails.vue';
  import { unnnicCallAlert } from '@weni/unnnic-system';

  import { mapActions } from 'vuex';
  import millify from 'millify';

  export default {
    name: 'AppPage',
    components: {
      Navigator,
      AppImagesBanner,
      AppDetailsHeader,
      AppDetailsAbout,
      // AppDetailsRecommended,
      AppDetailsComments,
      skeletonLoading,
    },
    data() {
      return {
        loading: true,
        app: null,
        bannerKey: 0,
      };
    },
    watch: {
      $route: 'fetchApp',
    },
    async mounted() {
      await this.fetchApp(this.$route.params.appCode);
    },
    methods: {
      ...mapActions(['getAppType', 'postRating']),
      async fetchApp(appCode) {
        const { data } = await this.getAppType(appCode);
        this.app = data;
        this.loading = false;
      },
      async reloadSection() {
        await this.fetchApp(this.$route.params.appCode);
        this.reloadBanner();
      },
      async handleRating(rate) {
        try {
          const data = {
            code: this.app.code,
            payload: {
              rate,
            },
          };
          await this.postRating(data);
        } catch (err) {
          unnnicCallAlert({
            props: {
              text: this.$t('app_details.status_error'),
              title: 'Error',
              icon: 'check-circle-1-1',
              scheme: 'feedback-red',
              position: 'bottom-right',
              closeText: this.$t('general.Close'),
            },
            seconds: 3,
          });
        } finally {
          this.reloadSection();
        }
      },
      reloadBanner() {
        this.bannerKey += 1;
      },
    },
    computed: {
      appRatingString() {
        return this.app.rating.average
          ? parseFloat(this.app.rating.average.toFixed(2)).toString()
          : '0';
      },
      appRatingAverage() {
        return this.app.rating.average ? parseFloat(this.app.rating.average.toFixed(2)) : 0;
      },
      navigatorHistory() {
        let history = [
          {
            name: this.$t('apps.details.navigator.discovery'),
            path: '/apps/discovery',
          },
        ];
        if (this.app) {
          history.push({
            name: this.app.name,
            path: '',
          });
        }
        return history;
      },
      appLinks() {
        const links = this.app.assets.filter((asset) => asset.type === 'LK');
        return links;
      },
      appMetrics() {
        return this.app.metrics ? millify(this.app.metrics) : '-';
      },
      appIntegrationsCount() {
        return this.app.integrations_count ? millify(this.app.integrations_count) : '-';
      },
    },
  };
</script>

<style scoped lang="scss">
  .app-details {
    &__section {
      margin: $unnnic-spacing-stack-md 0;

      &__columns {
        display: flex;

        &__recommended {
          max-width: 256px;
          margin-left: auto;
        }
      }
    }

    &__banner {
      ::v-deep .unnnic-banner-info__section {
        align-self: center;
      }
    }
  }
</style>
