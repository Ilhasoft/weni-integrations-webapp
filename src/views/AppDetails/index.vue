<template>
  <div>
    <navigator class="navigator" :routes="navigatorHistory" />
<<<<<<< HEAD

=======
>>>>>>> 4e067734185eee6ee14ddd4329b9599b20178800
    <div class="app-details" v-if="!loadingCurrentAppType">
      <app-images-banner
        class="app-details__section"
        :images="currentAppType && currentAppType.banners"
      />
      <app-details-header class="app-details__section" :app="currentAppType" />
      <unnnic-banner
        class="app-details__banner"
        :key="bannerKey"
        type="info"
        :firstTitle="$t('apps.details.info')"
        :firstDescription="appMetrics"
        :secondTitle="$t('apps.details.integrated_into')"
        :secondDescription="appIntegrationsCount"
        :subtitle="$t('apps.details.projects')"
        :thirdTitle="$t('apps.details.rating')"
        :thirdDescription="appRatingString"
        :rating="appRatingAverage"
        @ratingAction="handleRating"
      />
      <div class="app-details__section app-details__section__columns">
        <app-details-about :description="appDescription" :links="appLinks" />
      </div>
      <app-details-comments :appCode="appCode" />
    </div>
    <skeleton-loading v-else />
  </div>
</template>

<script>
  import Navigator from '../../components/Navigator/index.vue';
  import AppImagesBanner from '../../components/app/AppImagesBanner.vue';
  import AppDetailsHeader from '../../components/app/AppDetailsHeader.vue';
  import AppDetailsAbout from '../../components/app/AppDetailsAbout.vue';
  import AppDetailsComments from '../../components/app/AppDetailsComments.vue';
  import skeletonLoading from '../loadings/AppDetails.vue';
<<<<<<< HEAD
  import unnnicCallAlert from '@weni/unnnic-system';
=======
  import alert from '@/utils/call';
>>>>>>> 4e067734185eee6ee14ddd4329b9599b20178800
  import { app_type } from '@/stores/modules/appType/appType.store';
  import { mapActions, mapState } from 'pinia';
  import millify from 'millify';

  export default {
    name: 'AppPage',
    components: {
      Navigator,
      AppImagesBanner,
      AppDetailsHeader,
      AppDetailsAbout,
<<<<<<< HEAD
      // AppDetailsRecommended,
=======
>>>>>>> 4e067734185eee6ee14ddd4329b9599b20178800
      AppDetailsComments,
      skeletonLoading,
    },
    data() {
      return {
        bannerKey: 0,
      };
    },
    watch: {
      $route: 'fetchApp',
    },
    async mounted() {
      await this.fetchApp(this.$route.params.appCode);
    },
    methods: {
      ...mapActions(app_type, ['getAppType', 'postRating']),
      async fetchApp(appCode, shouldLoad = true) {
        await this.getAppType({ code: appCode, shouldLoad });
      },
      async reloadSection() {
        await this.fetchApp(this.$route.params.appCode, false);
        this.reloadBanner();
      },
      async handleRating(rate) {
        const data = {
          code: this.currentAppType.code,
          payload: {
            rate,
          },
        };
        await this.postRating(data);

        if (this.errorPostRating) {
<<<<<<< HEAD
          unnnicCallAlert({
            props: {
              text: this.$t('apps.details.status_error'),
              title: this.$t('general.error'),
              icon: 'check-circle-1-1',
              scheme: 'feedback-red',
              position: 'bottom-right',
              closeText: this.$t('general.Close'),
=======
          alert.callAlert({
            props: {
              text: this.$t('apps.details.status_error'),
              type: 'error',
>>>>>>> 4e067734185eee6ee14ddd4329b9599b20178800
            },
            seconds: 3,
          });
          return;
        }

        this.reloadSection();
      },
      reloadBanner() {
        this.bannerKey += 1;
      },
    },
    computed: {
      ...mapState(app_type, ['currentAppType', 'loadingCurrentAppType', 'appType.errorPostRating']),
      appRatingString() {
        return this.currentAppType?.rating?.average
          ? parseFloat(this.currentAppType.rating.average.toFixed(2)).toString()
          : '0';
      },
      appRatingAverage() {
        return this.currentAppType?.rating?.average
          ? parseFloat(this.currentAppType.rating.average.toFixed(2))
          : 0;
      },
      appDescription() {
        return this.currentAppType?.description;
      },
      navigatorHistory() {
        let history = [
          {
            name: this.$t('apps.details.navigator.discovery'),
            path: '/apps/discovery',
          },
        ];
        if (this.currentAppType?.name) {
          history.push({
            name: this.currentAppType.name,
            path: '',
          });
        }
        return history;
      },
      appLinks() {
        if (!this.currentAppType) {
          return [];
        }
        const links = this.currentAppType.assets.filter((asset) => asset.type === 'LK');
        return links;
      },
      appMetrics() {
        return this.currentAppType?.metrics ? millify(this.currentAppType.metrics) : '-';
      },
      appIntegrationsCount() {
        return this.currentAppType?.integrations_count
          ? millify(this.currentAppType.integrations_count)
          : '-';
      },
      appCode() {
        return this.currentAppType?.code;
      },
    },
  };
</script>

<style scoped lang="scss">
  @import './styles.scss';
</style>
app_type,
