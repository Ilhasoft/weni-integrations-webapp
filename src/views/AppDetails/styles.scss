.app-details {
    &__section {
      margin: $unnnic-spacing-stack-md 0;

      &__columns {
        display: flex;

        &__recommended {
          max-width: 256px;
          margin-left: auto;
        }
      }
    }

    &__banner {
      ::v-deep .unnnic-banner-info__section {
        align-self: center;
      }
    }
  }