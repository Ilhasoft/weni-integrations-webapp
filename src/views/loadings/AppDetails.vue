<template>
  <div class="loading-app-details">
    <unnnic-skeleton-loading tag="div" height="150px" width="100%" />

    <div class="loading-app-details__title">
      <unnnic-skeleton-loading tag="div" height="70px" width="70px" />
      <unnnic-skeleton-loading tag="div" height="48px" width="160px" />
    </div>

    <unnnic-skeleton-loading tag="div" height="130px" width="100%" />

    <div class="loading-app-details__description">
      <unnnic-skeleton-loading tag="div" height="207px" width="65vw" />
      <unnnic-skeleton-loading tag="div" height="207px" width="25vw" />
    </div>
    <!-- <app-images-banner class="app-details__section" :images="app.banners" />
    <app-details-header
      class="app-details__section"
      :title="app.name"
      :description="app.summary"
      :icon="app.icon"
      :iconbgColor="app.bg_color"
      :appCode="app.code"
    />
    <unnnic-banner
      :key="bannerKey"
      type="info"
      :firstTitle="$t('apps.details.info')"
      :firstDescription="appMetrics"
      :secondTitle="$t('apps.details.integrated_into')"
      :secondDescription="appIntegrationsCount"
      :subtitle="$t('apps.details.organizations')"
      :thirdTitle="$t('apps.details.rating')"
      :thirdDescription="appRatingString"
      :rating="appRatingAverage"
      @ratingAction="handleRating"
    />
    <div class="app-details__section app-details__section__columns">
      <app-details-about :description="app.description" :links="appLinks" />
      <app-details-recommended class="app-details__section__columns__recommended" />
    </div>
    <app-details-comments :appCode="app.code" /> -->
  </div>
</template>

<script>
  export default {};
</script>

<style lang="scss" scoped>
  .loading-app-details {
    width: 100%;
    margin-top: $unnnic-spacing-stack-xl;

    &__title {
      display: flex;
      justify-content: space-between;
      margin: $unnnic-spacing-stack-md 0;
    }

    &__description {
      display: flex;
      justify-content: space-between;
      margin-top: $unnnic-spacing-stack-xl;
    }
  }
</style>
