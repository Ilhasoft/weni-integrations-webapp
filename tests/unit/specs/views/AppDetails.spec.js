import { unnnicCallAlert as mockUnnnicCallAlert } from '@weni/unnnic-system';

jest.mock('@weni/unnnic-system', () => ({
  ...jest.requireActual('@weni/unnnic-system'),
  unnnicCallAlert: jest.fn(),
}));

jest.mock('@/api/appType', () => jest.fn());

import { shallowMount, createLocalVue } from '@vue/test-utils';
import AppDetails from '@/views/AppDetails.vue';
import i18n from '@/utils/plugins/i18n';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import { singleApp } from '../../../__mocks__/appMock';

const router = new VueRouter();

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);

describe('AppDetails.vue', () => {
  let wrapper;
  let store;
  let actions;
  let state;

  beforeEach(() => {
    actions = {
      getAppType: jest.fn(() => {
        return { data: singleApp };
      }),
      postRating: jest.fn(),
    };

    state = {
      appType: {
        currentAppType: singleApp,
        loadingCurrentAppType: false,
        errorCurrentAppType: false,
      },
    };

    store = new Vuex.Store({
      actions,
      state,
    });

    wrapper = shallowMount(AppDetails, {
      localVue,
      store,
      i18n,
      router,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        UnnnicBanner: true,
        Navigator: true,
        AppImagesBanner: true,
        AppDetailsHeader: true,
        AppDetailsAbout: true,
        AppDetailsRecommended: true,
        AppDetailsComments: true,
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('fetchApp()', () => {
    it('should call fetch app from API', async () => {
      expect(actions.getAppType).toHaveBeenCalledTimes(1);
      const appCode = 'code';
      await wrapper.vm.fetchApp(appCode);
      expect(actions.getAppType).toHaveBeenCalledTimes(2);
      expect(actions.getAppType).toHaveBeenCalledWith(expect.any(Object), {
        code: appCode,
        shouldLoad: true,
      });
    });
  });

  describe('appRatingString()', () => {
    it('should return zero if rating average is null', async () => {
      store.state.appType.currentAppType = { ...singleApp, rating: { average: null } };
      expect(wrapper.vm.appRatingString).toEqual('0');
      expect(typeof wrapper.vm.appRatingString).toEqual('string');
    });

    it('should return rating as a string if not null', async () => {
      store.state.appType.currentAppType = { ...singleApp, rating: { average: 2 } };
      expect(wrapper.vm.appRatingString).toEqual('2');
      expect(typeof wrapper.vm.appRatingString).toEqual('string');
    });
  });

  describe('appLinks()', () => {
    beforeEach(() => {
      wrapper.vm.app = {
        code: 'code',
        rating: { average: null },
        assets: [],
      };
    });

    it('should return no links', () => {
      const assets = [{ type: 'notLink' }, { type: 'notLink' }];
      wrapper.vm.app.assets = assets;
      expect(wrapper.vm.appLinks).toHaveLength(0);
    });

    it('should return valid links', () => {
      const assets = [{ type: 'notLink' }, { type: 'LK' }];
      store.state.appType.currentAppType = { ...singleApp, assets };
      expect(wrapper.vm.appLinks).toHaveLength(1);
    });
  });

  describe('navigatorHistory()', () => {
    it('should return history with current app', () => {
      expect(wrapper.vm.navigatorHistory).toHaveLength(2);
      expect(wrapper.vm.navigatorHistory[0].path).toEqual('/apps/discovery');
      expect(wrapper.vm.navigatorHistory[1].path).toEqual('');
    });

    it('should return history without current app', () => {
      store.state.appType.currentAppType = {};
      expect(wrapper.vm.navigatorHistory).toHaveLength(1);
      expect(wrapper.vm.navigatorHistory[0].path).toEqual('/apps/discovery');
    });
  });

  describe('handleRating()', () => {
    beforeEach(() => {
      wrapper.vm.app = {
        rating: { average: null },
        assets: [],
      };
    });

    it('should call postRating()', async () => {
      expect(actions.postRating).not.toHaveBeenCalled();
      const rate = 4;
      await wrapper.vm.handleRating(rate);
      expect(actions.postRating).toHaveBeenCalledTimes(1);
      expect(actions.postRating).toHaveBeenCalledWith(expect.any(Object), {
        code: wrapper.vm.currentAppType.code,
        payload: { rate },
      });
    });

    it('should call unnnicCallAlert on error', async () => {
      store.state.appType.errorPostRating = true;
      expect(mockUnnnicCallAlert).not.toHaveBeenCalled();
      await wrapper.vm.handleRating();
      expect(mockUnnnicCallAlert).toHaveBeenCalledTimes(1);
    });

    describe('appMetrics', () => {
      it('should return hifen if no app metrics', () => {
        expect(wrapper.vm.appMetrics).toEqual('-');
      });

      it('should return metric as string if exists', async () => {
        store.state.appType.currentAppType = { ...singleApp, metrics: 2 };
        expect(wrapper.vm.appMetrics).toEqual('2');
      });
    });

    describe('appIntegrationsCount', () => {
      it('should return hifen if no app integrations_count', () => {
        expect(wrapper.vm.appIntegrationsCount).toEqual('-');
      });

      it('should return integrations_count as string if exists', async () => {
        store.state.appType.currentAppType = { ...singleApp, integrations_count: 2 };
        expect(wrapper.vm.appIntegrationsCount).toEqual('2');
      });
    });
  });
});
