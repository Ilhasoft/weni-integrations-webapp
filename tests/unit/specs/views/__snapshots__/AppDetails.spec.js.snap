// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDetails.vue should be rendered properly 1`] = `
<div>
  <navigator-stub routes="[object Object],[object Object]" class="navigator"></navigator-stub>
  <div class="app-details">
    <appimagesbanner-stub class="app-details__section"></appimagesbanner-stub>
    <appdetailsheader-stub appcode="wwc" title="Weni Web Chat" description="Ullamco occaecat et id cillum." icon="https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/logo-wwc.svg" iconbgcolor="#00DED333" class="app-details__section"></appdetailsheader-stub>
    <unnnicbanner-stub type="info" firsttitle="some specific text" secondtitle="some specific text" thirdtitle="some specific text" firstdescription="-" seconddescription="-" thirddescription="4.9" subtitle="some specific text" rating="4.9"></unnnicbanner-stub>
    <div class="app-details__section app-details__section__columns">
      <appdetailsabout-stub description="Ullamco occaecat et id cillum. Amet exercitation nisi amet fugiat mollit minim est. Officia in enim amet ipsum Lorem velit sint pariatur sunt magna cupidatat. Magna non ea qui nisi ut.s" links=""></appdetailsabout-stub>
    </div>
    <appdetailscomments-stub appcode="wwc"></appdetailscomments-stub>
  </div>
</div>
`;
